packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -325,3 +315,94 @@ function getUpdatedWorkspaceWindowState(workspaceAtTop: OpenWorkspace) {
 export function resetWorkspaceStore() {
   getWorkspaceStore().setState(initialState);
 }
+
+type PromptType = 'closing-workspace' | 'closing-all-workspaces' | 'closing-workspace-launching-new-workspace';
+
+export function showWorkspacePrompts(
+  promptType: PromptType,
+  onConfirmation: () => void = () => {},
+  workspaceTitle: string = '',
+) {
+  const store = getWorkspaceStore();
+
+  switch (promptType) {
+    case 'closing-workspace': {
+      const prompt: Prompt = {
+        title: translateFrom('@openmrs/esm-patient-chart-app', 'unsavedChangesTitleText', 'Unsaved Changes'),
+        body: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'unsavedChangeText',
+          `You have unsaved changes in the side panel. Do you want to discard these changes?`,
+        ),
+        onConfirm: () => {
+          onConfirmation?.();
+        },
+        confirmText: translateFrom('@openmrs/esm-patient-chart-app', 'discard', 'Discard'),
+      };
+      store.setState({ ...store.getState(), prompt });
+      store.setState((prevState) => ({ ...prevState, prompt }));
+      return;
+    }
+
+    case 'closing-all-workspaces': {
+      const workspacesNotClosed = store
+        .getState()
+        .openWorkspaces.filter(({ name }) => !getWhetherWorkspaceCanBeClosed(name))
+        .map(({ title }, indx) => `${indx + 1}. ${title}`);
+
+      const prompt: Prompt = {
+        title: translateFrom('@openmrs/esm-patient-chart-app', 'unsavedChanges', 'You have unsaved changes'),
+        body: translateFrom(
+          '@openmrs/esm-patient-chart-app',
+          'unsavedChangesInForms',
```

Comment:
Why is this key called `unsavedChangesInForms` and not `unsavedChangesInWorkspaces`?

Reply:
I agree, will make the change